python/lsst/resources/_resourcePath.py

```diff
@@ -1360,7 +1402,7 @@ def open(
         required to reimplement `open`, though they may delegate to `super`
         when ``prefer_file_temporary`` is `False`.
         """
-        if self.dirLike:
+        if self.isdir():
             raise IsADirectoryError(f"Directory-like URI {self} cannot be opened.")
```

Comment:
You might consider removing this check entirely -- won't the underlying open() implementation throw an appropriate error if we try to open a directory as a file?

Reply:
Hopefully as_local is not a hotspot. 

* For `file` URIs this is a no-op so it will let a directory through and you'll get a deferred error sometime later when you call read().
* For S3 you will currently get a zero-byte file which will successfully be downloaded and which you can open.

So at minimum the dirLike test is needed (at least until S3 stops writing 0-byte directories), but given this is about to potentially involve downloading lots of data the isdir() check seems fairly harmless.